python/ray/experimental/data/dataset.py

```diff
@@ -31,6 +31,35 @@
 from ray.experimental.data.impl.arrow_block import (
     DelegatingArrowBlockBuilder, ArrowBlock)
 
+try:
+    import torch  # noqa: F811
+    from torch.utils.data import IterableDataset
+except ImportError:
+    torch = None
+    IterableDataset = None
+
+if IterableDataset is not None:
+
+    class RayIterableDataset(IterableDataset):
+        def __init__(self, generator_func):
+            self.generator_func = generator_func
```

Comment:
maybe instead of a conditional definition, you can just define this in another file and import it inline in `to_torch`?

Reply:
good point, done